acquia/acquia_cms
# ACMS-1073: Add settings to enable Headless robust api capabilities

**Motivation**
Need to provide a user a starting point for partially / fully headless sites.  

**Proposed changes**
Adds a new service to the ACMS Headless module that when enabled via the Robust API form, completes the following:
- Creates a Headless user Role
- Creates a new Header user
- Creates a consumer entity (with a secret key provided to the user)
- Creates an OAUTH public and key key pair.
- Creates a Next.js site entity with Preview Secret Key
- Gathers a list of available node entities and creates a list of next.js site entity types.
- Also updates the headless.settings config with the base headless User & Consumer UUIDs.

**Alternatives considered**
N / A

**Testing steps**
- Pull branch down locally.
- Do a fresh install of an ACMS site.
- Enable the Acquia CMS Headless module.
- Navigate to the tour page
- enable the robust api option and save
- copy down the secret provided on screen.
- verify that there's a new Headless User and Role, Headless site 1 consumer (default) and Next Js site, as well as Next js entity types matching the available content types.
- Verify the consumer secret is stored, hashed in the database (`drush sql-query "select label, secret from consumer_field_data;"`
- After enabling ACMS Headless and the Robust API feature, see the readme for additional steps for setting up a next js site.

**Merge requirements**
